src/planner/binder/expression/bind_lambda.cpp

```diff
@@ -1,38 +1,159 @@
+		lambda_bindings->push_back(new_lambda_binding);
+
+		// bind the parameter expressions
+		for (idx_t i = 0; i < expr.params.size(); i++) {
+			auto result = BindExpression(&expr.params[i], depth, false);
+			D_ASSERT(!result.HasError());
+		}
+
+		auto result = BindExpression(&expr.expr, depth, false);
+		lambda_bindings->pop_back();
+
+		// successfully bound a subtree of nested lambdas, set this to nullptr in case other parts of the
+		// query also contain lambdas
+		if (lambda_bindings->empty()) {
+			lambda_bindings = nullptr;
+		}
+
+		// now bind the rhs as a normal expression
+		return result;
+	}
+
+	// this is for binding JSON
+	auto lhs_expr = expr.lhs->Copy();
+	OperatorExpression arrow_expr(ExpressionType::ARROW, move(lhs_expr), move(expr.expr));
+	return BindExpression(arrow_expr, depth);
+}
+
+void ExpressionBinder::TransformLambdaExprChild(unique_ptr<Expression> &original, unique_ptr<Expression> &replacement,
+                                                vector<unique_ptr<Expression>> &arguments,
+                                                LogicalType &list_child_type) {
+
+	// check if the original expression is a lambda parameter
+	bool is_lambda_parameter = false;
+	if (original->expression_class == ExpressionClass::BOUND_COLUMN_REF) {
+
+		// determine if this is the lambda parameter
+		auto &bound_col_ref = (BoundColumnRefExpression &)*original;
+		if (bound_col_ref.binding.table_index == DConstants::INVALID_INDEX) {
+			is_lambda_parameter = true;
+		}
+	}
+
+	if (is_lambda_parameter) {
+		// this is a lambda parameter, so the replacement refers to the first argument, which is the list
+		replacement = make_unique<BoundReferenceExpression>(arguments[0]->alias, list_child_type, 0);
+
+	} else {
+		// this is not a lambda parameter, so we need to create a new argument for the arguments vector
+		replacement = make_unique<BoundReferenceExpression>(original->alias, original->return_type, arguments.size());
+		arguments.push_back(move(original));
+	}
+}
+
+void ExpressionBinder::IterateLambdaExprChildren(vector<unique_ptr<Expression>> &arguments,
+                                                 LogicalType &list_child_type, unique_ptr<Expression> &expr) {
+
+	if (expr->expression_class == ExpressionClass::BOUND_SUBQUERY) {
+		throw InvalidInputException("Subqueries are not supported in lambda expressions!");
+	}
+
+	// these expression classes do not have children, transform them
+	if (expr->expression_class == ExpressionClass::BOUND_CONSTANT ||
+	    expr->expression_class == ExpressionClass::BOUND_COLUMN_REF ||
+	    expr->expression_class == ExpressionClass::BOUND_DEFAULT ||
+	    expr->expression_class == ExpressionClass::BOUND_PARAMETER ||
+	    expr->expression_class == ExpressionClass::BOUND_REF) {
+
+		// move the expr because we are going to replace it
+		auto original = move(expr);
+		unique_ptr<Expression> replacement;
+
+		TransformLambdaExprChild(original, replacement, arguments, list_child_type);
```

Comment:
Should we really do this for all these types (`BOUND_REF, BOUND_PARAMETER, BOUND_CONSTANT, BOUND_DEFAULT`)? Should this not only happen with `column_ref`? Is this tested with the other types?